chef_master/source/custom_resources.rst

```diff
@@ -817,6 +818,128 @@ For example, the following properties define ``username`` and ``password`` prope
+
+       .. code-block:: ruby
+
+          property :name, String
+
+       .. code-block:: ruby
+
+          property :name, Fixnum
+
+       .. code-block:: ruby
+
+          property :name, Hash
+
+       .. code-block:: ruby
+
+          property :name, [TrueClass, FalseClass]
+
+       .. code-block:: ruby
+
+          property :name, [String, NilClass]
+
+       .. code-block:: ruby
+
+          property :name, [Class, String, Symbol]
+
+       .. code-block:: ruby
+
+          property :name, [Array, Hash]
+
+.. end_tag
+
+validators
++++++++++++++++++++++++++++++++++++++++++++++++++++++
+.. tag dsl_custom_resource_method_property_validation_parameter
+
+A validation parameter is used to add zero (or more) validation parameters to a property.
+
+.. list-table::
+   :widths: 150 450
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``:callbacks``
+     - Use to define a collection of unique keys and values (a ruby hash) for which the key is the error message and the value is a lambda to validate the parameter. For example:
+
+       .. code-block:: ruby
+  
+          callbacks: {
+                       'should be a valid non-system port' => lambda {
+                         |p| p > 1024 && p < 65535
+                       }
+                     }
+
+   * - ``:default``
+     - Use to specify the default value for a property. For example:
+
+       .. code-block:: ruby
+       
+          default: 'a_string_value'
+       
+       .. code-block:: ruby
+       
+          default: 123456789
+       
+       .. code-block:: ruby
+       
+          default: []
+       
+       .. code-block:: ruby
+       
+          default: ()
+       
+       .. code-block:: ruby
+       
+          default: {}
+   * - ``:equal_to``
+     - Use to match a value with ``==``. Use an array of values to match any of those values with ``==``. For example:
+
+       .. code-block:: ruby
+
+          equal_to: [true, false]
+
+       .. code-block:: ruby
+          
+          equal_to: ['php', 'perl']
+   * - ``:regex``
+     - Use to match a value to a regular expression. For example:
+
+       .. code-block:: ruby
+       
+          regex: [ /^([a-z]|[A-Z]|[0-9]|_|-)+$/, /^\d+$/ ]
+   * - ``:required``
+     - Indicates that a property is required. For example:
+
+       .. code-block:: ruby
+       
+          required: true
+   * - ``:respond_to``
+     - Use to ensure that a value has a given method. This can be a single method name or an array of method names. For example:
+
+       .. code-block:: ruby
+       
+          respond_to: valid_encoding?
+
+Some examples of combining validation parameters:
+
+.. code-block:: ruby
+
+   property :spool_name, String, regex: /$\w+/
+
+.. code-block:: ruby
+
+   property :enabled, equal_to: [true, false, 'true', 'false'], default: true
```

Comment:
Still missing a ruby type here, shouldn't it be ```property :enabled, [TrueClass, FalseClass, String], equal_to: [true, false, 'true', 'false'], default: true``` According to the above data (or if the type can be omitted, this worth a note even if I think it's a bad idea to not specify things)